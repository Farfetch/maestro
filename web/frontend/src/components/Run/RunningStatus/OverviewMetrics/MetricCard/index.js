import MetricCard from "./MetricCard";

export default MetricCard;
