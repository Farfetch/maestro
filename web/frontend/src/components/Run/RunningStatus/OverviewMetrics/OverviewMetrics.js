/* eslint-disable max-statements */
import { Descriptions } from "antd";
import React from "react";

import { avg } from "../../../../lib/utils";
import MetricCard from "./MetricCard";

const OverviewMetrics = ({ metrics, totals }) => (
  <div
    style={{
      border: "1px solid #fff "
    }}
  >
    <Descriptions>
      <div
        style={{
          flexWrap: "wrap"
        }}
      >
        <Descriptions.Item>
          <MetricCard
            title="Average Latency"
            value={avg(totals.totalLatencyAvg, metrics.length).toFixed(2)}
            unit="ms"
            borderColor="#ffd166"
          />
        </Descriptions.Item>
        <Descriptions.Item>
          <MetricCard
            title="90% Latency"
            value={avg(totals.totalLatencyP90, metrics.length).toFixed(2)}
            unit="ms"
            borderColor="#ffab40"
          />
        </Descriptions.Item>
        <Descriptions.Item>
          <MetricCard
            title="Errors"
            value={totals.totalErrorsRate}
            unit="%"
            borderColor="#ff6b6b"
          />
        </Descriptions.Item>
        <Descriptions.Item>
          <MetricCard
            title="Average RPM"
            value={avg(parseFloat(totals.totalRpm), metrics.length).toFixed(0)}
            unit="req/min"
            borderColor="#64b5f6"
          />
        </Descriptions.Item>
      </div>
    </Descriptions>
  </div>
);

export default OverviewMetrics;
