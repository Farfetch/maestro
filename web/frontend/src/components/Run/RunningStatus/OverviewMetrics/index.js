import OverviewMetrics from "./OverviewMetrics";

export default OverviewMetrics;
