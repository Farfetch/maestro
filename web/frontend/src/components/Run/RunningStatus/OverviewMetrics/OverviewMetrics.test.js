import { render } from "@testing-library/react";
import React from "react";

import OverviewMetrics from "./OverviewMetrics";

describe("OverviewMetrics", () => {
  const mockMetrics = [
    {
      key: "metric1",
      errorsCount: 5,
      successCount: 20,
      errorRate: 25,
      label: "Label 1",
      rpm: 50,
      totalCount: 25,
      latencyAvg: 50,
      latencyP50: 45,
      latencyP90: 60,
      latencyP99: 75,
      responses: [{ responseCode: "200" }, { responseCode: "404" }]
    },
    {
      key: "metric2",
      errorsCount: 3,
      successCount: 15,
      errorRate: 20,
      label: "Label 2",
      rpm: 60,
      totalCount: 18,
      latencyAvg: 55,
      latencyP50: 50,
      latencyP90: 65,
      latencyP99: 80,
      responses: [{ responseCode: "200" }, { responseCode: "500" }]
    }
  ];

  const mockTotals = {
    totalLatencyAvg: 105,
    totalLatencyP50: 95,
    totalLatencyP90: 125,
    totalLatencyP99: 155,
    totalSuccessCount: 35,
    totalErrorsCount: 8,
    totalErrorsRate: 22.86,
    totalCount: 43,
    totalRpm: 110
  };

  test("should render OverviewMetrics component", () => {
    const rendered = render(
      <OverviewMetrics
        metrics={mockMetrics}
        totals={mockTotals}
        isLoading={false}
      />
    );
    const component = rendered.container;
    expect(component.outerHTML).toMatchSnapshot();
  });
});
