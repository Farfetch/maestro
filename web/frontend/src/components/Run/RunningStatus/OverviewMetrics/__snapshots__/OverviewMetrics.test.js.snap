// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewMetrics should render OverviewMetrics component 1`] = `"<div><div style=\\"border: 1px solid #fff;\\"><div class=\\"ant-descriptions\\"><div class=\\"ant-descriptions-view\\"><table><tbody><tr class=\\"ant-descriptions-row\\"><td class=\\"ant-descriptions-item\\" style=\\"flex-wrap: wrap;\\" colspan=\\"3\\"><div class=\\"ant-descriptions-item-container\\"><span class=\\"ant-descriptions-item-content\\"><div class=\\"ant-statistic\\" style=\\"border: 1px solid #ffd166; border-radius: 10px; box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.2); flex: 1 1 auto; margin: 8px; text-align: center; padding: 5px;\\"><div class=\\"ant-statistic-title\\">Average Latency</div><div class=\\"ant-statistic-content\\"><span class=\\"ant-statistic-content-value\\"><span class=\\"ant-statistic-content-value-int\\">52</span><span class=\\"ant-statistic-content-value-decimal\\">.50</span></span><span class=\\"ant-statistic-content-suffix\\">ms</span></div></div><div class=\\"ant-statistic\\" style=\\"border: 1px solid #ffab40; border-radius: 10px; box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.2); flex: 1 1 auto; margin: 8px; text-align: center; padding: 5px;\\"><div class=\\"ant-statistic-title\\">90% Latency</div><div class=\\"ant-statistic-content\\"><span class=\\"ant-statistic-content-value\\"><span class=\\"ant-statistic-content-value-int\\">62</span><span class=\\"ant-statistic-content-value-decimal\\">.50</span></span><span class=\\"ant-statistic-content-suffix\\">ms</span></div></div><div class=\\"ant-statistic\\" style=\\"border: 1px solid #ff6b6b; border-radius: 10px; box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.2); flex: 1 1 auto; margin: 8px; text-align: center; padding: 5px;\\"><div class=\\"ant-statistic-title\\">Errors</div><div class=\\"ant-statistic-content\\"><span class=\\"ant-statistic-content-value\\"><span class=\\"ant-statistic-content-value-int\\">22</span><span class=\\"ant-statistic-content-value-decimal\\">.86</span></span><span class=\\"ant-statistic-content-suffix\\">%</span></div></div><div class=\\"ant-statistic\\" style=\\"border: 1px solid #64b5f6; border-radius: 10px; box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.2); flex: 1 1 auto; margin: 8px; text-align: center; padding: 5px;\\"><div class=\\"ant-statistic-title\\">Average RPM</div><div class=\\"ant-statistic-content\\"><span class=\\"ant-statistic-content-value\\"><span class=\\"ant-statistic-content-value-int\\">55</span></span><span class=\\"ant-statistic-content-suffix\\">req/min</span></div></div></span></div></td></tr></tbody></table></div></div></div></div>"`;
